oauth2client/client.py

```diff
@@ -841,9 +785,8 @@ def _refresh(self, http_request):
         if not self.store:
             self._do_refresh_request(http_request)
         else:
-            self.store.acquire_lock()
-            try:
-                new_cred = self.store.locked_get()
+            with self.store.lock():
```

Comment:
Yeah this is a work in progress, I haven't quite decided on the behavior here. Will follow up in a top-level comment.